app/helpers/application_helper.rb

```diff
@@ -53,7 +53,7 @@ def date_time_relative_value(time)
   end
 
   def iana_timezone
-    ActiveSupport::TimeZone::MAPPING[Time.zone.try(:name)] || 'UTC'
+    ActiveSupport::TimeZone::MAPPING.has_value?(Time.zone.try(:name)) ? Time.zone.try(:name) : ActiveSupport::TimeZone::MAPPING[Time.zone.try(:name)] || 'UTC'
   end
```

Comment:
@ezr-ondrej : Thanks for the pointer about the cookies. I was not aware of that functionality.
So upon debugging, I realized that the cookie timezone was already in iana tz format. 
Therefore,  ` ActiveSupport::TimeZone::MAPPING[Time.zone.try(:name)] ` was returning nil and timezone was being set to UTC.
Added a check to see if the Time.zone has already flowed from the cookie in iana format and returning that if that's the case.

Reply:
Cool catch!